calicoctl/commands/ipam/show.go

```diff
@@ -24,80 +24,136 @@ import (
+	bapi "github.com/projectcalico/libcalico-go/lib/backend/api"
 )
 
-// IPAM takes keyword with an IP address then calls the subcommands.
-func Show(args []string) error {
-	doc := constants.DatastoreIntro + `Usage:
-  calicoctl ipam show [--ip=<IP> | --show-blocks] [--config=<CONFIG>]
+type borrowedIP struct {
+	addr           string
+	borrowingNode  string
+	block          string
+	blockOwner     string
+	allocationType string
+	allocatedTo    string
+}
 
-Options:
-  -h --help             Show this screen.
-     --ip=<IP>          Report whether this specific IP address is in use.
-     --show-blocks      Show detailed information for IP blocks as well as pools.
-  -c --config=<CONFIG>  Path to the file containing connection configuration in
-                        YAML or JSON format.
-                        [default: ` + constants.DefaultConfigPath + `]
+func getPoolDetails(ctx context.Context, ippoolClient clientv3.IPPoolInterface, bc bapi.Client) ([]*borrowedIP, error) {
+	var details []*borrowedIP
 
-Description:
-  The ipam show command prints information about a given IP address, or about
-  overall IP usage.
-`
-	parsedArgs, err := docopt.Parse(doc, args, true, "", false, false)
+	// Read details for all blocks.
+	blocks, err := bc.List(ctx, model.BlockListOptions{}, "")
 	if err != nil {
-		return fmt.Errorf("Invalid option: 'calicoctl %s'. Use flag '--help' to read about a specific subcommand.", strings.Join(args, " "))
-	}
-	if len(parsedArgs) == 0 {
-		return nil
+		return nil, err
 	}
+	for _, kvp := range blocks.KVPairs {
+		b := kvp.Value.(*model.AllocationBlock)
 
-	ctx := context.Background()
+		for i := range b.Allocations {
+			if b.Allocations[i] != nil {
+				// Allocations[i] if not nil:
+				// - b.OrdinalToIP(i) is the allocated IP address
+				// - *b.Allocations[i] is the index of the corresponding b.Attributes
 
-	// Create a new backend client from env vars.
-	cf := parsedArgs["--config"].(string)
-	client, err := clientmgr.NewClient(cf)
+				attributes := b.Attributes[*b.Allocations[i]]
+
+				if b.Affinity != nil {
+					if borrowingNode, ok := attributes.AttrSecondary[model.IPAMBlockAttributeNode]; ok {
+
+						// Affinity is in the form host:mgianluc-bz-09s4-kadm-node-2
+						// Remove "host:"
+						var host string
+						parts := strings.Split(*b.Affinity, ":")
+						if len(parts) == 2 {
+							host = parts[1]
+						} else {
+							host = parts[0]
+						}
+
+						if host != borrowingNode {
+							bIP := borrowedIP{block: b.CIDR.IPNet.String(), blockOwner: host, borrowingNode: borrowingNode}
+							bIP.addr = b.OrdinalToIP(i).String()
+							if _, ok := attributes.AttrSecondary[model.IPAMBlockAttributePod]; ok {
+								bIP.allocatedTo = fmt.Sprintf("%s/%s", attributes.AttrSecondary[model.IPAMBlockAttributeNamespace],
+									attributes.AttrSecondary[model.IPAMBlockAttributePod])
+								bIP.allocationType = "model.IPAMBlockAttributePod"
```

Comment:
Since this is user facing, I think it would be better to just set the type to "Pod" rather than "model.IPAMBlockAttributePod", since users don't know what that is